### DISCUSSION TITLE
Speeding Up Single File App Decompression

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/runtime#96772] [Q&A]
I have a .NET 7x app that compiles as a single file app with the following characteristics:
```
SelfContained = true,
EnableCompressionInSingleFile = true,
PublishSingleFile = true,
IncludeNativeLibrariesForSelfExtract = true,
IncludeAllContentForSelfExtract = true,
```

My compiled EXE is about 250mb and the decompressed files are about 750mb.

When running my app, on first launch, it might take as long as 30 seconds for the UI to be displayed and this is predominantly because of the decompression process.  This is particularly confusing to users because there is no splash screen or loading screen.  I dont think this is possible because any code I were to write would not be run until after the EXE decompresses.

What options are available so that I can inform my customer that my app has launched while it is decompressing?

**Comment by alexandrehtrb** [comment]
You could pack your program, not single-file published and not compressed, in an installer.

The installer then would place these files in a Program Files folder and the program could be started from a desktop shortcut.

**Comment by vitek-karas** [comment]
Related https://github.com/dotnet/runtime/issues/50081.
